Geometry/GEMGeometryBuilder/src/GEMGeometryParsFromDD.cc

```diff
@@ -29,86 +29,183 @@ GEMGeometryParsFromDD::~GEMGeometryParsFromDD()
+  
+  DDBooleanSolid solid = (DDBooleanSolid)(fv.logicalPart().solid());
+  std::vector<double> dpar = solid.solidA().parameters();
+  
+  double dy = dpar[0]/cm;//length is along local Y
+  double dz = dpar[3]/cm;// thickness is long local Z
+  double dx1= dpar[4]/cm;// bottom width is along local X
+  double dx2= dpar[8]/cm;// top width is along local X
+  dpar = solid.solidB().parameters();
+  dz += dpar[3]/cm;// chamber thickness
+
+  GEMDetId gemid = detId.chamberId();
+  std::string name = fv.logicalPart().name().name();
+  std::vector<std::string> strpars{name};
+
+  std::vector<double> pars{dx1, dx2, dy, dz};
+  std::vector<double> vtra = getTranslation(fv);
+  std::vector<double> vrot = getRotation(fv);
+  
+  rgeo.insert(gemid.rawId(), vtra, vrot, pars, strpars);
+}
 
-  while (doSubDets){
+void
+GEMGeometryParsFromDD::buildEtaPartition(DDFilteredView& fv, GEMDetId detId, RecoIdealGeometry& rgeo)
+{
+  LogDebug("GEMGeometryParsFromDD") << "buildEtaPartition "<<fv.logicalPart().name().name()
+				    <<" "<< detId <<std::endl;
+  
+  // EtaPartition specific parameter (nstrips and npads) 
+  DDValue numbOfStrips("nStrips");
+  DDValue numbOfPads("nPads");
+  std::vector<const DDsvalues_type* > specs(fv.specifics());
+  std::vector<const DDsvalues_type* >::iterator is = specs.begin();
+  double nStrips = 0., nPads = 0.;
+  for (;is != specs.end(); is++){
+    if (DDfetch( *is, numbOfStrips)) nStrips = numbOfStrips.doubles()[0];
+    if (DDfetch( *is, numbOfPads))   nPads = numbOfPads.doubles()[0];
+  }
+  LogDebug("GEMGeometryParsFromDD") 
+    << ((nStrips == 0. ) ? ("No nStrips found!!") : ("Number of strips: " + std::to_string(nStrips))); 
+  LogDebug("GEMGeometryParsFromDD") 
+    << ((nPads == 0. ) ? ("No nPads found!!") : ("Number of pads: " + std::to_string(nPads)));
+  
+  // EtaPartition specific parameter (size) 
+  std::vector<double> dpar = fv.logicalPart().solid().parameters();
 
-    // Get the Base Muon Number
-    MuonDDDNumbering mdddnum(muonConstants);
-    MuonBaseNumber   mbn = mdddnum.geoHistoryToBaseNumber(fview.geoHistory());
+  double dy = dpar[0]/cm;//length is along local Y
+  double dz = 0.4/cm;// thickness is long local Z
+  double dx1= dpar[4]/cm;// bottom width is along local X
+  double dx2= dpar[8]/cm;// top width is along local X
 
-    // Get the The GEM det Id 
-    GEMNumberingScheme gemnum(muonConstants);
-    int detid = 0;
+  std::string name = fv.logicalPart().name().name();
+  std::vector<std::string> strpars{name};
 
-    detid = gemnum.baseNumberToUnitNumber(mbn);
-    GEMDetId gemid(detid);
+  std::vector<double> pars{dx1, dx2, dy, dz, nStrips, nPads};
+  std::vector<double> vtra = getTranslation(fv);
+  std::vector<double> vrot = getRotation(fv);
+  
+  rgeo.insert(detId.rawId(), vtra, vrot, pars, strpars);
+}
 
-    DDValue numbOfStrips("nStrips");
-    DDValue numbOfPads("nPads");
+std::vector<double> GEMGeometryParsFromDD::getTranslation(DDFilteredView& fv)
```

Comment:
this and getRotation should return a std::array (or an actual vector/matrix object that would probably be more appropriate..)

Reply:
RecoIdealGeometry wants std::vectors. If I make this into std::array, I would need to do an std::array to std::vector conversion.